Source/WebCore/dom/ScriptExecutionContext.cpp

```diff
@@ -957,6 +959,21 @@ RefCountedSerialFunctionDispatcher& ScriptExecutionContext::nativePromiseDispatc
     return *m_nativePromiseDispatcher;
 }
 
+bool ScriptExecutionContext::requiresScriptExecutionTelemetry(ScriptTelemetryCategory category)
+{
+    Ref vm = protectedVM();
+    if (!vm->topCallFrame)
+        return false;
+
+    auto [taintedness, taintedURL] = JSC::sourceTaintedOriginFromStack(vm, vm->topCallFrame);
+
+    // FIXME: Use this information to emit a console log message.
+    UNUSED_PARAM(taintedURL);
+    UNUSED_PARAM(category);
+
+    return taintedness == JSC::SourceTaintedOrigin::KnownTainted;
```

Comment:
This function isn’t used right now, but in the future won’t we also want this to return true when `taintedness == JSC::SourceTaintedOrigin::IndirectlyTainted`?

Reply:
Ah, I left it `KnownTainted` for now to be a bit more conservative, but I can go ahead and include this now.